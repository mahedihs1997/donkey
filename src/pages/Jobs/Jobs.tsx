import VirtualizedTable from '@components/VirtualizedTable';
import { Box, Button, Grid, Paper, Typography } from '@material-ui/core';
import React from 'react';

import MockJobs from '../../mocks/jobs.json';

const Jobs = () => {
  return (
    <div>
      <Grid direction="row" container={true}>
        <Typography variant="h4" style={{ marginRight: 20 }}>
          Jobs
        </Typography>
        <Button type="button" variant="contained" color="primary">
          Add new Job
        </Button>
      </Grid>
      <Box mt={5} />
      <Paper style={{ height: 800, maxWidth: 1000 }}>
        <VirtualizedTable
          rowCount={MockJobs.length}
          rowGetter={({ index }) => MockJobs[index]}
          columns={[
            {
              width: 120,
              label: 'Job Number',
              dataKey: 'jobNumber',
            },
            {
              width: 120,
              label: 'Customer Id',
              dataKey: 'customerId',
            },
            {
              width: 120,
              label: 'Customer Name',
              dataKey: 'customerName',
            },
            {
              width: 120,
              label: 'Date Posted',
              dataKey: 'date',
            },
            {
              width: 120,
              label: 'Vehicle',
              dataKey: 'vehicle',
            },
            {
              width: 120,
              label: 'Model',
              dataKey: 'model',
            },
            {
              width: 120,
              label: 'Status',
              dataKey: 'status',
            },
            {
              width: 120,
              label: 'Service Type',
              dataKey: 'status',
            },
          ]}
        />
      </Paper>
    </div>
  );
};

Jobs.getInitialProps = async () => {
  return {
    showLayout: true,
    title: 'Jobs',
  };
};

export default Jobs;
