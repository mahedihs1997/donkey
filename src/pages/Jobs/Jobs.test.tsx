import React from 'react';
import TestRenderer from 'react-test-renderer';

import Jobs from '.';

describe('<Jobs />', () => {
  it('should render without throwing an error and match snapshot', () => {
    const testRenderer = TestRenderer.create(<Jobs />);
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
