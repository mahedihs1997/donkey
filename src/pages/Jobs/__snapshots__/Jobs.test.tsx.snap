// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Jobs /> should render without throwing an error and match snapshot 1`] = `
<div>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4"
      style={
        Object {
          "marginRight": 20,
        }
      }
    >
      Jobs
    </h4>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Add new Job
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    className="MuiBox-root MuiBox-root-1"
  />
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
    style={
      Object {
        "height": 800,
        "maxWidth": 1000,
      }
    }
  >
    <div
      style={
        Object {
          "height": 0,
          "overflow": "visible",
          "width": 0,
        }
      }
    >
      <div
        aria-colcount={8}
        aria-rowcount={50}
        className="ReactVirtualized__Table MuiVirtualizedTable-table-3"
        role="grid"
        style={Object {}}
      >
        <div
          className="ReactVirtualized__Table__headerRow MuiVirtualizedTable-tableRow-4 MuiVirtualizedTable-flexContainer-2"
          role="row"
          style={
            Object {
              "height": 48,
              "overflow": "hidden",
              "paddingRight": 0,
              "width": 0,
            }
          }
        >
          <div
            className="ReactVirtualized__Table__headerColumn"
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 120px",
                "flex": "0 1 120px",
                "msFlex": "0 1 120px",
              }
            }
          >
            <div
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiVirtualizedTable-tableCell-6 MuiVirtualizedTable-flexContainer-2 MuiVirtualizedTable-noClick-7 MuiTableCell-alignLeft"
              role="cell"
              style={
                Object {
                  "height": 48,
                }
              }
            >
              <span>
                Job Number
              </span>
            </div>
          </div>
          <div
            className="ReactVirtualized__Table__headerColumn"
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 120px",
                "flex": "0 1 120px",
                "msFlex": "0 1 120px",
              }
            }
          >
            <div
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiVirtualizedTable-tableCell-6 MuiVirtualizedTable-flexContainer-2 MuiVirtualizedTable-noClick-7 MuiTableCell-alignLeft"
              role="cell"
              style={
                Object {
                  "height": 48,
                }
              }
            >
              <span>
                Customer Id
              </span>
            </div>
          </div>
          <div
            className="ReactVirtualized__Table__headerColumn"
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 120px",
                "flex": "0 1 120px",
                "msFlex": "0 1 120px",
              }
            }
          >
            <div
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiVirtualizedTable-tableCell-6 MuiVirtualizedTable-flexContainer-2 MuiVirtualizedTable-noClick-7 MuiTableCell-alignLeft"
              role="cell"
              style={
                Object {
                  "height": 48,
                }
              }
            >
              <span>
                Customer Name
              </span>
            </div>
          </div>
          <div
            className="ReactVirtualized__Table__headerColumn"
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 120px",
                "flex": "0 1 120px",
                "msFlex": "0 1 120px",
              }
            }
          >
            <div
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiVirtualizedTable-tableCell-6 MuiVirtualizedTable-flexContainer-2 MuiVirtualizedTable-noClick-7 MuiTableCell-alignLeft"
              role="cell"
              style={
                Object {
                  "height": 48,
                }
              }
            >
              <span>
                Date Posted
              </span>
            </div>
          </div>
          <div
            className="ReactVirtualized__Table__headerColumn"
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 120px",
                "flex": "0 1 120px",
                "msFlex": "0 1 120px",
              }
            }
          >
            <div
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiVirtualizedTable-tableCell-6 MuiVirtualizedTable-flexContainer-2 MuiVirtualizedTable-noClick-7 MuiTableCell-alignLeft"
              role="cell"
              style={
                Object {
                  "height": 48,
                }
              }
            >
              <span>
                Vehicle
              </span>
            </div>
          </div>
          <div
            className="ReactVirtualized__Table__headerColumn"
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 120px",
                "flex": "0 1 120px",
                "msFlex": "0 1 120px",
              }
            }
          >
            <div
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiVirtualizedTable-tableCell-6 MuiVirtualizedTable-flexContainer-2 MuiVirtualizedTable-noClick-7 MuiTableCell-alignLeft"
              role="cell"
              style={
                Object {
                  "height": 48,
                }
              }
            >
              <span>
                Model
              </span>
            </div>
          </div>
          <div
            className="ReactVirtualized__Table__headerColumn"
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 120px",
                "flex": "0 1 120px",
                "msFlex": "0 1 120px",
              }
            }
          >
            <div
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiVirtualizedTable-tableCell-6 MuiVirtualizedTable-flexContainer-2 MuiVirtualizedTable-noClick-7 MuiTableCell-alignLeft"
              role="cell"
              style={
                Object {
                  "height": 48,
                }
              }
            >
              <span>
                Status
              </span>
            </div>
          </div>
          <div
            className="ReactVirtualized__Table__headerColumn"
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 120px",
                "flex": "0 1 120px",
                "msFlex": "0 1 120px",
              }
            }
          >
            <div
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiVirtualizedTable-tableCell-6 MuiVirtualizedTable-flexContainer-2 MuiVirtualizedTable-noClick-7 MuiTableCell-alignLeft"
              role="cell"
              style={
                Object {
                  "height": 48,
                }
              }
            >
              <span>
                Service Type
              </span>
            </div>
          </div>
        </div>
        <div
          aria-label="grid"
          aria-readonly={null}
          className="ReactVirtualized__Grid ReactVirtualized__Table__Grid"
          onScroll={[Function]}
          role="rowgroup"
          style={
            Object {
              "WebkitOverflowScrolling": "touch",
              "boxSizing": "border-box",
              "direction": "inherit",
              "height": -48,
              "overflowX": "hidden",
              "overflowY": "auto",
              "position": "relative",
              "width": 0,
              "willChange": "transform",
            }
          }
          tabIndex={0}
        />
      </div>
    </div>
  </div>
</div>
`;
